### ISSUE TITLE
Cookie set with path != "/" cannot be retrieved

### ISSUE THREAD

**Issue by masenf** [reflex-dev/reflex#1628]
**Describe the bug**
Cookies accept a `path` argument that restricts them to a subpath of the domain. When navigating directly to a subpath in a reflex app and setting a cookie, the cookie gets the subpath by default and then cannot be used from other pages in the app (or at all?).

Additionally, the path that reflex uses for get/set cookie only seems to be updated when refreshing or initially loading the page, but _not when navigating between pages_ with `rx.redirect` or `rx.link`

**To Reproduce**
```python
import reflex as rx

COOKIE = "cookie"


class State(rx.State):
    @rx.var
    def cookie_value(self) -> str:
        return self.get_cookies().get(COOKIE, "Not set")

    @rx.var
    def path(self) -> str:
        return self.get_current_page() or "/"


def index() -> rx.Component:
    return rx.vstack(
        rx.heading("You are viewing '", State.path, "'", font_size="2em"),
        rx.box("The cookie value is ", rx.code(State.cookie_value)),
        rx.button("Set Cookie", on_click=rx.set_cookie(COOKIE, "🍪")),
        rx.button("Remove Cookie", on_click=rx.remove_cookie(COOKIE)),
        rx.hstack(
            rx.link("/", href="/"),
            rx.link("/foo", href="/foo"),
            padding_top="2em",
        ),
        padding_top="10%",
    )


# Add state and page to the app.
app = rx.App()
app.add_page(index)
app.add_page(index, route="/foo")
app.compile()
```

1. Load the root of the app
2. Click "Set Cookie"
3. Refresh the page, notice that cookie is now set. Navigate to `/foo` and refresh the page, notice the cookie is still set.
4. Click "Remove Cookie" from the `/foo` page and refresh the page. Notice that the cookie is still set.
5. Navigate back to the root of the app, and refresh. Click "Remove Cookie" and refresh again. Notice that the cookie shows as `Not set`.
6. Navigate to the `/foo` page and refresh. Click "Set Cookie" from the `/foo` page and refresh the page again. Notice that the cookie still shows as `Not set`.
7. Navigate back to the root of the app and refresh the page; still `Not set`.
8. Open the browser dev tools, Storage tab, and examine the cookies. Notice that the cookie is actually set, but due to the `path` argument it does not seem usable.

The refreshes are very important to reproducing the behavior. If one loads the root and navigates to `/foo` without refreshing, then things do seem to work, which makes this bug particularly hard to characterize and reproduce (i.e. loss of sanity ensue).

**Expected behavior**
Set a cookie and read it back or remove it from any page in the app.

**Screenshots**
Check it:

https://github.com/reflex-dev/reflex/assets/1524005/fee7b4f8-c433-4b3e-a66a-af2fca49d891

**Specifics (please complete the following information):**
 - Python Version: 3.11.4
 - Reflex Version: 0.2.5a1
 - OS: macOS 13.5.1
 - Browser (Optional): Firefox 116, Safari 16.6
 
**Additional context**
Since reflex does NOT specify `{path: "/"}` when setting the cookie, browser default behavior takes over:

> ;path=path: Indicates the path that must exist in the requested URL for the browser to send the [Cookie](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cookie) header (e.g., '/', '/mydir'). If not specified, it defaults to the current path of the current document location. [[1]](https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie)

https://github.com/reflex-dev/reflex/blob/f5067d4050c4cff7e5435754ccab33721a4f1556/reflex/.templates/web/utils/state.js#L127-L130

**Comment by Lendemor** [comment]
Is this covered by https://github.com/reflex-dev/reflex/pull/1629 ?

**Comment by masenf** [comment]
fixed via #1629
